da/celestia/celestia.go

```diff
@@ -159,62 +174,63 @@ func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 			c.logger.Debug("Context cancelled")
 			return da.ResultSubmitBatch{}
 		default:
-			txResponse, err := c.client.SubmitPFB(c.ctx, c.config.NamespaceID, blob, c.config.Fee, c.config.GasLimit)
-			if txResponse != nil {
-				if txResponse.Code != 0 {
-					c.logger.Debug("Failed to submit DA batch. Emitting health event and trying again", "txResponse", txResponse, "error", err)
-					// Publish an health event. Only if we failed to emit the event we return an error.
-					res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, false, errors.New(txResponse.RawLog))
-					if err != nil {
-						return res
-					}
-				} else if err != nil {
-					// Here we assume that if txResponse is not nil and also error is not nil it means that the transaction
-					// was submitted (not necessarily accepted) and we still didn't get a clear status regarding it (e.g timeout).
-					// hence trying to poll for it.
-					c.logger.Debug("Failed to receive DA batch inclusion result. Waiting for inclusion", "txResponse", txResponse.RawLog, "error", err)
-					inclusionHeight, err := c.waitForTXInclusion(txResponse.TxHash)
-					if err == nil {
-						res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, true, nil)
-						if err != nil {
-							return res
-						} else {
-							return da.ResultSubmitBatch{
-								BaseResult: da.BaseResult{
-									Code:     da.StatusSuccess,
-									Message:  "tx hash: " + txResponse.TxHash,
-									DAHeight: inclusionHeight,
-								},
-							}
-						}
-					} else {
-						c.logger.Debug("Failed to receive DA batch inclusion result. Emitting health event and trying again", "error", err)
-						res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, false, err)
-						if err != nil {
-							return res
-						}
-					}
-
-				} else {
-					c.logger.Debug("Successfully submitted DA batch", "txResponse", txResponse.RawLog)
-					res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, true, nil)
-					if err != nil {
-						return res
-					}
-					return da.ResultSubmitBatch{
-						BaseResult: da.BaseResult{
-							Code:     da.StatusSuccess,
-							Message:  "tx hash: " + txResponse.TxHash,
-							DAHeight: uint64(txResponse.Height),
-						},
-					}
+			estimatedGas := float64(EstimateGas([]int{len(blob)}))
+			gasWanted := uint64(estimatedGas * gasAdjustment)
+			fees := c.calculateFees(gasWanted)
+
+			//SubmitPFB sets an error if the txResponse has error, so we check check the txResponse for error
+			txResponse, err := c.client.SubmitPFB(c.ctx, c.config.NamespaceID, blob, int64(fees), gasWanted)
+			if txResponse == nil {
```

Comment:
seems like the only difference between this `if` and the one beneath is the error returned and log.
Possibly can combine them both in the same `if`, e.g `if txResponse == nil || txResponse.Code != 0` and then simply create the relevant error.